Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
 def is_white_row(view, points):
   return all(view.substr(pt).isspace() for pt in reversed(points))
 
-def get_breadcrumb(view, points, limit):
+def get_breadcrumb(view, points, regex, separator):
   for pt in points:
     ch = view.substr(pt)
     if not ch.isspace():
-      return view.substr(sublime.Region(pt, min(view.line(pt).b, pt + limit))).strip()
+      linestring = view.substr(sublime.Region(pt, min(view.line(pt).b, pt + 500))).strip()
+      match = re.search(re.compile(regex), linestring)
+      if match:
+        return(separator + match.group('name'))
   return ''
 
-class BreadcrumbsCommand(sublime_plugin.EventListener):
-  def on_selection_modified(self, view):
-    tab_size = get_tab_size(view)
-    settings = sublime.load_settings('Breadcrumbs.sublime-settings')
-    breadcrumb_length_limit = settings.get('breadcrumb_length_limit', 100)
-    separator = settings.get('breadcrumbs_separator', u' ■ ')
-    total_breadcrumbs_length_limit = settings.get('total_breadcrumbs_length_limit', 200)
-    view.erase_status('breadcrumbs')
-
-    if len(view.sel()) == 0:
-      return
+def make_breadcrumbs(view):
+  tab_size = get_tab_size(view)
+  my_regex = settings.get('breadcrumbs_regex', u'(?P<name>.*)')
```

Comment:
> I'm not a native speaker, in case you haven't noticed

Me neither, probably why I didn't notice. What was that quote again about programming and naming things? ;)

Singular is better here, agreed.